test/persistence-hooks.suite.js

```diff
@@ -1828,6 +2068,442 @@ module.exports = function(dataSource, should) {
       });
     });
 
+    describe('PersistedModel.replaceOrCreate', function() {
+      it('triggers hooks in the correct order on create', function(done) {
+        monitorHookExecution();
+
+        TestModel.replaceOrCreate(
+          { id: 'not-found', name: 'not found' },
+          function(err, record, created) {
+            if (err) return done(err);
+            triggered.should.eql([
+              'access',
+              'before save',
+              'persist',
+              'loaded',
+              'after save'
+            ]);
+            done();
+          });
+      });
+
+      it('triggers hooks in the correct order on replace', function(done) {
+        monitorHookExecution();
+
+        TestModel.replaceOrCreate(
+          { id: existingInstance.id, name: 'new name' },
+          function(err, record, created) {
+            if (err) return done(err);
+            if (dataSource.connector.replaceOrCreate) {
+              triggered.should.eql([
+                'access',
+                'before save',
+                'persist',
+                'loaded',
+                'after save'
+              ]); 
+            } else {
+              triggered.should.eql([
+                'access',
+                'loaded',
```

Comment:
>As I commented above, since findOne is called with options.notify:false, I would expect it does not trigger the loaded hook?

Very true! You are right thats a bug in `find` which gets called by `findOne; in `find` [irrespective of whether `options.notify`is false or not](https://github.com/strongloop/loopback-datasource-juggler/blob/master/lib/dao.js#L1578-L1583) it called [allcb](https://github.com/strongloop/loopback-datasource-juggler/blob/master/lib/dao.js#L1495-L1576) where it triggers all hooks; I guess it makes sense if I do not touch it in this PR... It should be done in another PR... thought?